src/Symfony/Component/Validator/Constraints/JsonValidator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ */
+class JsonValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Json) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Json');
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!$this->isJson($value)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setCode(Json::NOT_JSON_ERROR)
+                ->addViolation();
+        }
+    }
+
+    /**
+     * @param string $string
+     *
+     * @return bool
+     */
+    protected function isJson($string)
```

Comment:
`isJson(string $string): bool` which makes the docblock obsolete. Not sure we really need this extension point? could be private / inlined into `validate()`.